matrix/common/etl/transformers/specimen_library.py

```diff
@@ -30,49 +42,125 @@ def _write_rows_to_psvs(self, *args: typing.Tuple):
     def _parse_from_metadatas(self, search_dir: str):
```

Comment:
Not crucial for this PR, but we should definitely refactor the metadata transformers (excludes `cell_expression` transformers which technically parse more than metadata) to favor modularity over performance. Transformer groups were initially defined by the metadata files they consumed in order to avoid processing the same file multiple times (e.g. ontology determination in the `specimen` table depends on `library` metadata creating `specimen_library.py`).

Favoring modularity proposes defining a transformer per table. Considering the limited number of rows we can expect per project for `specimen`, `donor`, `library`, `project`, `publication` and `contributor` entities, processing these entities multiple times will not affect performance. I propose writing one transformer for each of these entities, as well as `cell_suspension` (which will be processed once per the table structure in this PR), to improve readability of the code without affecting performance.

Reply:
Yeah I agree, this class is now kind of a random hodgepodge of code. It will probably get worse if we need to start creating more "derived" fields.

Reply:
[Ticket](https://app.zenhub.com/workspaces/dcp-5ac7bcf9465cb172b77760d9/issues/humancellatlas/matrix-service/420). Resolving.